addons/website_sale/tests/test_website_sale_combo_configurator.py

```diff
@@ -0,0 +1,116 @@
+        no_variant_attribute = self.env['product.attribute'].create({
+            'name': "No variant attribute",
+            'create_variant': 'no_variant',
+            'value_ids': [
+                Command.create({'name': "A"}),
+                Command.create({'name': "B", 'is_custom': True, 'default_extra_price': 1}),
+            ],
+        })
+        product_a1 = self.env['product.template'].create({
+            'name': "Product A1",
+            'attribute_line_ids': [
+                Command.create({
+                    'attribute_id': no_variant_attribute.id,
+                    'value_ids': [Command.set(no_variant_attribute.value_ids.ids)],
+                }),
+            ],
+        })
+        combo_a = self.env['product.combo'].create({
+            'name': "Combo A",
+            'combo_item_ids': [
+                Command.create({'product_id': product_a1.product_variant_id.id, 'extra_price': 5}),
+                Command.create({'product_id': self._create_product(name="Product A2").id}),
+            ],
+        })
+        combo_b = self.env['product.combo'].create({
+            'name': "Combo B",
+            'combo_item_ids': [
+                Command.create({'product_id': self._create_product(name="Product B1").id}),
+                Command.create({'product_id': self._create_product(name="Product B2").id}),
+            ],
+        })
+        self._create_product(
+            name="Combo product",
+            list_price=25,
+            type='combo',
+            combo_ids=[
+                Command.link(combo_a.id),
+                Command.link(combo_b.id),
+            ],
+            website_published=True,
+        )
+        self.start_tour('/', 'website_sale_combo_configurator', login='admin')
+
+    def test_website_sale_combo_configurator_single_configuration(self):
+        """ Test that the combo configurator isn't shown if there's a single configuration. """
+        no_variant_attribute = self.env['product.attribute'].create({
+            'name': "No variant attribute",
+            'create_variant': 'no_variant',
+            'value_ids': [Command.create({'name': "A"})],
+        })
+        product = self.env['product.template'].create({
+            'name': "Test product",
+            'attribute_line_ids': [
+                Command.create({
+                    'attribute_id': no_variant_attribute.id,
+                    'value_ids': [Command.set(no_variant_attribute.value_ids.ids)],
+                }),
+            ],
+        })
+        combo = self.env['product.combo'].create({
+            'name': "Test combo",
+            'combo_item_ids': [Command.create({'product_id': product.product_variant_id.id})],
+        })
+        self._create_product(
+            name="Combo product",
+            type='combo',
+            combo_ids=[Command.link(combo.id)],
+            website_published=True,
+        )
```

Comment:
it is the same as in the next function, is it maybe better to extract it in the class setup ` setUpClass(cls)`   ?

Reply:
It's debatable :smile: Moving shared setup into `setUpClass` removes some duplication, but it doesn't necessarily make the test easier to understand. A good practice is to keep the setup which is directly related to what you're testing in the test method, and move boilerplate code that isn't directly related to the test into `setUpClass` or helper methods. As a side note, code duplication should be avoided in "normal" code, but it isn't always a bad thing in tests, where we aim to make each test self-explanatory (i.e. it shouldn't be necessary to dive into `setUpClass` or helper methods to understand the test).

For combo tests, it's a bit tricky to determine which part of the setup is boilerplate and which isn't. In this particular test, we need to ensure that the combo product isn't configurable, so the fact that
- We have a combo product with a single combo,
- Which has a single combo item,
- That references a non-configurable product,

Is fairly related to the test IMO. On the other hand, I can probably extract at least some of it into a helper method (e.g. to setup the combo product, combo, and combo item via a single method call). But if you don't mind, I'd rather do that in a follow-up PR :slightly_smiling_face: 

FYI, the setup is very similar but the attribute value in the method below is custom :)

Reply:
Do not mind at all, thank you for your explanation!